src/main/java/tasks/Deadline.java

```diff
@@ -0,0 +1,47 @@
+package tasks;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task with a specific due date and time.
+ * A deadline task includes a description and a due date/time by which the task should be completed.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime dueWhen;
+
+    /**
+     * Constructs a deadline task with the specified description and due date/time.
+     *
+     * @param description the description of the deadline task
+     * @param dueWhen the due date and time of the deadline task
+     */
+    public Deadline(String description, LocalDateTime dueWhen) {
+        super(description);
+        this.dueWhen = dueWhen;
+    }
+
+    /**
+     * Returns the string representation of the deadline task for display purposes.
+     * The format includes the status, description, and due date/time.
+     *
+     * @return the string representation of the deadline task
+     */
+    @Override
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
+        return "D | " + this.getStatusIcon() + " | " + this.description + " (by: " + this.dueWhen.format(formatter) + ")";
+    }
+
+    /**
+     * Returns the string representation of the deadline task suitable for saving to a file.
+     * The format is machine-readable, including the status, description, and due date/time.
+     *
+     * @return the string representation of the deadline task for file storage
+     */
+    @Override
+    public String toFileString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
```

Comment:
Same as the previous. Since this is a different format from the one used during runtime, it would help to create two separate constants for both saving format and runtime format, within the task class (maybe SAVING_DATE_TIME_FORMAT and RUNTIME_DATE_TIME_FORMAT). But nice job differentiating the two formats using javadocs to help me understand better what each separate format is used for :)!